datajoint/fetch.py

```diff
@@ -20,46 +25,57 @@ def prepare_attributes(relation, item):
         raise DataJointError("Index must be a slice, a tuple, a list, a string.")
     return item, attributes
 
-class FetchQuery:
+def copy_first(f):
+    @wraps(f)
+    def ret(*args, **kwargs):
+        args = list(args)
+        args[0] = args[0].__class__(args[0]) # call copy constructor
+        return f(*args, **kwargs)
 
-    def __init__(self, relation):
-        """
+    return ret
 
-        """
-        self.behavior = dict(
-            offset=0, limit=None, order_by=None, descending=False, as_dict=False, map=None
-        )
-        self._relation = relation
+class Fetch:
+    def __init__(self, relation):
+        if isinstance(relation, Fetch): # copy constructor
+            self.behavior = dict(relation.behavior)
+            self._relation = relation._relation
+        else:
+            self.behavior = dict(
+                offset=0, limit=None, order_by=None, as_dict=False
+            )
+            self._relation = relation
 
 
+    @copy_first
     def from_to(self, fro, to):
         self.behavior['offset'] = fro
```

Comment:
I am fine with offset_by  and limit_to.


Am 25. Juli 2015 00:28:19 schrieb Dimitri Yatsenko <notifications@github.com>:

> >
> >
> > +    @copy_first
> >      def from_to(self, fro, to):
> >          self.behavior['offset'] = fro
>
> Python index convention does not use the language 'from' and 'to', which 
> implies including the first and last values.  I really think we should 
> revert to SQL's `offset` and `limit`.  We don't need to reinvent the wheel. 
>  These are not used too often.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/datajoint/datajoint-python/pull/135/files#r35480056